src/stan/error_handling/out_of_range.hpp

```diff
@@ -0,0 +1,97 @@
+#ifndef STAN__ERROR_HANDLING__OUT_OF_RANGE_HPP
+#define STAN__ERROR_HANDLING__OUT_OF_RANGE_HPP
+
+#include <typeinfo>
+#include <string>
+#include <sstream>
+#include <stdexcept>
+#include <stan/meta/traits.hpp>
+
+namespace stan {
+  namespace error_handling {
+
+    /**
+     * Throw an out_of_range exception with a consistently formatted message.
+     * 
+     * This is an abstraction for all Stan functions to use when throwing
+     * out of range. This will allow us to change the behavior for all
+     * functions at once.
+     *
+     * The message is:
```

Comment:
Cutting and pasting the messages everywhere seems like it'll make things hard to refactor.  Can you just point somewhere to the formatting instead?